cscs-checks/tools/profiling_and_debugging/berkeleylab-roofline.py

```diff
@@ -0,0 +1,116 @@
+        self.sourcesdir = os.path.join(self.current_system.resourcesdir,
+                                       'roofline', 'ert')
+        self.build_system = 'SingleSource'
+        self.sourcepath = 'kernel1.c driver1.c'
+        self.executable = 'ert.exe'
+        self.build_system.ldflags = ['-O3 -fopenmp']
+        self.sourcesdir = os.path.join(self.current_system.resourcesdir,
+                                       'roofline', 'ert')
+        self.rpt = '%s.rpt' % self.executable
+        self.maintainers = ['JG']
+        self.tags = {'scs'}
+
+    def setup(self, partition, environ, **job_opts):
+        super().setup(partition, environ, **job_opts)
+        if self.num_tasks != 36:
+            self.job.launcher.options = ['--cpu-bind=verbose,none']
+
+
+@rfm.parameterized_test(
+    *[[mpitask, flop]
+    for mpitask in [36, 18, 12, 9, 6, 4, 3, 2, 1]
+    for flop in [1, 2, 4, 8, 16, 32, 64, 128, 256, 512, 1024]])
+class ErtBroadwellTest(ErtTestBase):
+    def __init__(self, mpitask, flop):
+        super().__init__()
+        ompthread = 36 // mpitask
+        self.valid_systems = ['daint:mc', 'dom:mc']
+        self.valid_prog_environs = ['PrgEnv-gnu']
+        self.build_system.cppflags = [
+            '-DERT_ALIGN=32',
+            '-DERT_MEMORY_MAX=1073741824',
+            '-DERT_MPI=True',
+            '-DERT_OPENMP=True',
+            '-DERT_TRIALS_MIN=1',
+            '-DERT_WORKING_SET_MIN=1',
+            '-DERT_FLOP=%s' % flop,
+        ]
+        self.name = 'ert_FLOPS.{:04d}_MPI.{:03d}_OpenMP.{:03d}'.format(
+            flop, mpitask, ompthread)
+        self.exclusive = True
+        self.num_tasks = mpitask
+        self.num_tasks_per_node = mpitask
+        self.num_cpus_per_task = ompthread
+        self.num_tasks_per_core = 1
+        self.use_multithreading = False
+        self.variables = {
+            'CRAYPE_LINK_TYPE': 'dynamic',
+            'OMP_NUM_THREADS': str(self.num_cpus_per_task)
+        }
+        # Reference roofline boundaries for Intel Broadwell CPU (E5-2695 v4):
+        GFLOPs = 945.0
+        L1bw = 1788.0
+        L2bw = 855.0
+        L3bw = 547.0
+        DRAMbw = 70.5
+        # slowest job:
+        mpitaskm1 = 1
+        flopm1 = 1024
+        self.roofline_rpt = 'rpt'
+        if mpitask == mpitaskm1 and flop == flopm1:
+            self.post_run = [
+                'cat *_job.out | python2 preprocess.py > pre',
+                'python2 maximum.py < pre > max',
+                'python2 summary.py < max > sum',
+                # give enough time for all the dependent jobs to collect data
+                'sleep 60',
+                'cat ../ert_FLOPS*/sum |python2 roofline.py > rpt',
+            ]
+            self.sanity_patterns = sn.all([
+                # --- check data type:
+                sn.assert_eq(sn.extractsingle(
+                    r'^\s+(?P<prec>\w+) \* __restrict__ buf = \(\w+ \*\)'
+                    r'malloc\(PSIZE\);', 'driver1.c', 'prec'), 'double'),
```

Comment:
Are you missing an argument to `assert_eq` here or is there sth wrong with the alignment?